File: src/main/java/FirstBadVersion/BBIGBOSS/FirstBadVersion.cpp
//
// Created by BBIGBOSS on 2019-06-17.
//

// Forward declaration of isBadVersion API.
bool isBadVersion(int version);

class Solution
{
    public:

    int firstBadVersion(int n)
    {
        long left = 1, right = n;

        while (left < right)
        {
            long mid = (left + right) / 2;

            if (!isBadVersion(mid))
                left = mid + 1;
            else
                right = mid;
        }

        return left;
    }
};

File: src/main/java/FirstBadVersion/codeLove/FirstBadVersion.java
package FirstBadVersion.codeLove;

public class FirstBadVersion {
    public int firstBadVersion(int n){

        int left=1,right=n;

        while( left <= right ){
            int mid=left+ ( right-left )/2;//防止N足够大时right+left溢出
            if( isBadVersion(mid) ){
                right = mid - 1 ;
            } else {
                left = mid + 1;
            }
        }
        return left;
    }
    public static boolean isBadVersion( int n ){
        return true;
    }
}
